src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientOptionsAnalyzer.cs

```diff
@@ -41,12 +44,47 @@ private void AnalyzeClientOptionsType(SymbolAnalysisContext context)
         {
             var typeSymbol = (INamedTypeSymbol)context.Symbol;
 
-            var serviceVersion = typeSymbol.GetTypeMembers("ServiceVersion");
+            var serviceVersion = typeSymbol.GetTypeMembers(ServiceVersionName);
             var serviceVersionEnum = serviceVersion.SingleOrDefault(member => member.TypeKind == TypeKind.Enum);
             if (serviceVersionEnum == null)
             {
                 context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0008, typeSymbol.Locations.First()));
+                return;
             }
+
+            foreach (var constructor in typeSymbol.Constructors)
+            {
+                if (constructor.DeclaredAccessibility == Accessibility.Public)
+                {
+                    if (constructor.Parameters == null || constructor.Parameters.Length == 0)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0009, constructor.Locations.First()));
+                        continue;
+                    }
+
+                    var firstParam = constructor.Parameters.FirstOrDefault();
+                    if (!IsServiceVersionParameter(firstParam))
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0009, firstParam.Locations.First()));
+                        continue;
+                    }
+
+                    if (!firstParam.HasExplicitDefaultValue)
```

Comment:
What do you think about requiring the highest enum value as part of AZC0010?

Reply:
Yes, this could work ... do you think there would ever be a situation where an unsupported version was added to the enum before it was officially supported?  

The guideline is "DO call the highest supported service API version by default," so it seems like that would be possible -- if there was a preview service release that we wanted to enable but not support.  LMK what you think.

Reply:
My general approach is to:
1. Be as strict as possible and see if it falls apart
2. Diagnostics can be disabled with `#pragma disable warning` for special cases.